userspace/libsinsp/filter/ast.h

```diff
@@ -26,6 +26,14 @@ namespace libsinsp {
 namespace filter {
 namespace ast {
 
+/*!
+	\brief Methods to escape/unescape strings
+	\note Throws a sinsp_exception in case of parsing errors.
+	\return an escaped/unescaped verison of the string
+*/
+std::string escape_str(const std::string& str);
```

Comment:
I would prefer having these as static methods of the `libsinsp::filter::parser` class. Effectively, this logic is dictated by the parsing grammar and is not tied to the AST structures. This would also have the parsing logic all in the same place.

Reply:
I was running into circular dependency problems with that, as ast was being included by parser. But I will declare standalone stuff to make it work.

Reply:
Oh I didn't think of that. In that case creating separate .h and .cpp files could be a solution maybe.

Reply:
I just declared escape_str standalone in ast.cpp.